src/ContractEditor/plugins/clausePlugin.js

```diff
@@ -54,29 +50,16 @@ function ClausePlugin() {
       clauseNode = editor.value.document.getParent(para.key);
     }
 
-    if (!clauseNode) {
+    if (!clauseNode || clauseNode.type !== 'clause') {
       return next();
     }
 
-    const template = clauseNode.data.get('template');
-
-    if (template) {
-      console.log(`Found template on node: ${template.getIdentifier()}`);
-      try {
-        const ciceroClause = new Clause(template);
-        ciceroClause.parse(textNode.text.trim());
-        const parseResult = ciceroClause.getData();
-        console.log(parseResult);
-        editor.command(GoodParse, clauseNode, parseResult);
-      } catch (error) {
-        console.log(error);
-        editor.command(BadParse, clauseNode, error);
-      }
-    } else {
-      console.log('Template not found on node.');
-    }
-
+    const nodeAttributes = clauseNode.data.get('attributes');
+    const { src, clauseid } = nodeAttributes;
 
+    parseClause(src, textNode.text.trim(), clauseid)
+      .then(parseResult => console.log(parseResult)) // add/remove annotation
```

Comment:
Not sure I entirely understand what `parseClause` is doing here, do we want it to be logging the result to the console?

Reply:
The comment is what we actually want in the future (to add or remove an annotation via slate). We'll be implementing new functions to add annotations once Slate pushes a fix to that bug that Dan was experiencing.